tests/runtime/flavorful/host.ts

```diff
@@ -1,53 +1,49 @@
-    testwasi,
-    imports: {
-      fListInRecord1(x) {},
-      fListInRecord2() { return { a: 'list_in_record2' }; },
-      fListInRecord3(x) {
-        assert.strictEqual(x.a, 'list_in_record3 input');
-        return { a: 'list_in_record3 output' };
-      },
-      fListInRecord4(x) {
-        assert.strictEqual(x.a, 'input4');
-        return { a: 'result4' };
-      },
-      fListInVariant1(a, b, c) {
-        assert.strictEqual(a, 'foo');
-        assert.deepStrictEqual(b, { tag: 'err', val: 'bar' });
-        assert.deepStrictEqual(c, { tag: 0, val: 'baz' });
-      },
-      fListInVariant2() { return 'list_in_variant2'; },
-      fListInVariant3(x) {
-        assert.strictEqual(x, 'input3');
-        return 'output3';
-      },
-
-      errnoResult() { return { tag: 'err', val: "b" }; },
-      listTypedefs(x, y) {
-        assert.strictEqual(x, 'typedef1');
-        assert.deepStrictEqual(y, ['typedef2']);
-        return [(new TextEncoder).encode('typedef3'), ['typedef4']];
-      },
+// Imports
+export function fListInRecord1(x: any) {}
+export function fListInRecord2() { return { a: 'list_in_record2' }; }
+export function fListInRecord3(x: any) {
+  assert.strictEqual(x.a, 'list_in_record3 input');
+  return { a: 'list_in_record3 output' };
+}
+export function fListInRecord4(x: any) {
+  assert.strictEqual(x.a, 'input4');
+  return { a: 'result4' };
+}
+export function fListInVariant1(a: any, b: any, c: any) {
+  assert.strictEqual(a, 'foo');
+  assert.deepStrictEqual(b, { tag: 'err', val: 'bar' });
+  assert.deepStrictEqual(c, { tag: 0, val: 'baz' });
+}
+export function fListInVariant2() { return 'list_in_variant2'; }
+export function fListInVariant3(x: any) {
+  assert.strictEqual(x, 'input3');
+  return 'output3';
+}
+export function errnoResult() { return { tag: 'err', val: "b" }; }
+export function listTypedefs(x: any, y: any) {
+  assert.strictEqual(x, 'typedef1');
+  assert.deepStrictEqual(y, ['typedef2']);
+  return [(new TextEncoder).encode('typedef3'), ['typedef4']];
+}
+export function listOfVariants(bools: any, results: any, enums: any) {
+  assert.deepStrictEqual(bools, [true, false]);
+  assert.deepStrictEqual(results, [{ tag: 'ok', val: undefined }, { tag: 'err', val: undefined }]);
+  assert.deepStrictEqual(enums, ["success", "a"]);
+  return [
+    [false, true],
+    [{ tag: 'err', val: undefined }, { tag: 'ok', val: undefined }],
+    ["a", "b"],
+  ];
+}
 
-      listOfVariants(bools, results, enums) {
-        assert.deepStrictEqual(bools, [true, false]);
-        assert.deepStrictEqual(results, [{ tag: 'ok', val: undefined }, { tag: 'err', val: undefined }]);
-        assert.deepStrictEqual(enums, ["success", "a"]);
-        return [
-          [false, true],
-          [{ tag: 'err', val: undefined }, { tag: 'ok', val: undefined }],
-          ["a", "b"],
-        ];
-      },
-    },
-  });
+export async function run () {
+  // @ts-ignore
+  const wasm = await import('./flavorful.js');
```

Comment:
Is this possible to lift to a top-level `import`?

Additionally, how come this needs `@ts-ignore`?

Reply:
This can't be lifted as it would create a top-level await execution cycle since the module imports the JS which imports this module (to fulfill the imports). We could just create another module, but it's quite nice to encapsulate the test in a single file as well.

Reply:
Turns out the `@ts-ignore` wasn't necessary.